src/main/java/sergey/EquationServlet.java

```diff
@@ -0,0 +1,194 @@
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws IOException {
+        try {
+            HttpSession session = req.getSession();
+//            System.out.println("DoGet " + req.getPathInfo() + " with:");
+            List<String> tokens = parseIntoPolishNotation(session.getAttribute("equation").toString());
+            Map<String, Integer> args = new HashMap<>();
+            Map<String, String> StringValue = new HashMap<>();
+            Enumeration<String> names = session.getAttributeNames();
+            while (names.hasMoreElements()) {
+                String key = names.nextElement();
+                if (key.length() != 1) continue;
+                String value = session.getAttribute(key).toString();
+                try {
+                    args.put(key, Integer.parseInt(value));
+                } catch (NumberFormatException e) {
+                    StringValue.put(key, value);
+                }
+            }
+            for (String key : StringValue.keySet()) {
+                Integer value = args.get(StringValue.get(key));
+                args.put(key, value);
+            }
+            int result = calculate(tokens, args);
+
+//            System.out.println("args: " + args.toString());
+//            System.out.println("result: " + result);
+
+            resp.setStatus(SC_OK);
+            resp.getOutputStream().print(result);
+            resp.getOutputStream().flush();
+            resp.getOutputStream().close();
+        } catch (ParseException e) {
+            resp.sendError(SC_CONFLICT, "Problem with equation");
+        } catch (NumberFormatException ee) {
+            resp.sendError(SC_CONFLICT, "Problem with variable");
+        } catch (NullPointerException eee) {
+            resp.sendError(SC_CONFLICT, "Not set required parameters");
+        }
+    }
+
+    private List<String> parseIntoPolishNotation(String equation) throws ParseException {
+        if (!(equation.contains("+")
+                || equation.contains("-")
+                || equation.contains("*")
+                || equation.contains("/"))) {
+            throw new ParseException("notValidEquation", 0);
+        }
+        List<String> tokens = new ArrayList<>();
+        StringTokenizer tokenizer = new StringTokenizer(equation.replaceAll(" ", ""), "()+-/*", true);
+        while (tokenizer.hasMoreElements()) {
+            tokens.add(tokenizer.nextToken());
+        }
+        Deque<String> stack = new ArrayDeque<>();
+        List<String> polishNotation = new ArrayList<>();
+        for (String token : tokens) {
+            try {
+                switch (token) {
+                    case "+":
+                    case "-":
+                    case ")":
+                        while ("+-/*".contains(stack.getFirst())) {
+                            polishNotation.add(stack.removeFirst());
+                        }
+                        if (token.equals(")")) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Position literals first in String comparisons](https://app.codacy.com/app/thejerome/IFMO_Servlets_20190915/pullRequest?prid=4723796)